### PR TITLE
feat: add pubsublite-kafka-auth module

### PR DESCRIPTION
**Pull request by dpcollins-google**
This allows for auth through kafka params.

### CODE REVIEW

**Review by anguillanneuf — COMMENTED** [review]
I did a pass. Other than the inline comments, could you add a short description for the Python script either in the README or in the file itself? 

You probably need to break out the samples into a different PR after releasing the library with the change.

File: pubsublite-kafka-auth/pom.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <groupId>com.google.cloud</groupId>
+    <artifactId>pubsublite-kafka-parent</artifactId>
+    <version>1.0.3-SNAPSHOT</version><!-- {x-version-update:pubsublite-kafka:current} -->
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>pubsublite-kafka-auth</artifactId>
+  <version>1.0.3-SNAPSHOT</version><!-- {x-version-update:pubsublite-kafka:current} -->
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply pubsublite-kafka-auth/pom.xml:11]: Yeah this is fine

File: samples/snippets/src/main/java/pubsublite/KafkaProducerExample.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Google LLC
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply samples/snippets/src/main/java/pubsublite/KafkaProducerExample.java]: Done.

File: samples/snippets/src/main/java/pubsublite/KafkaProducerExample.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+
+public class KafkaProducerExample {
+
+  public static void main(String... args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String cloudRegion = "us-central1";
+    char zoneId = 'a';
+    // Use an existing Pub/Sub Lite topic.
+    String topicId = "dpcollins-beam-schema-test";
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply samples/snippets/src/main/java/pubsublite/KafkaProducerExample.java]: Done.

File: samples/snippets/src/main/java/pubsublite/KafkaProducerExample.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    // Using the project number is required for constructing a Pub/Sub Lite
+    // topic path that the Kafka producer can use.
+    long projectNumber = Long.parseLong("129988248131");
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply samples/snippets/src/main/java/pubsublite/KafkaProducerExample.java]: Done.

File: pubsublite-kafka-auth/src/main/java/com/google/cloud/pubsublite/kafka/ClientParameters.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    params.put("sasl.jaas.config", getJaasConfig(project));
+    params.put(
+        "sasl.login.callback.handler.class",
+        "org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerLoginCallbackHandler");
+    return params;
+  }
+
+  private static String getEndpoint(CloudRegion region) {
+    return region.value() + "-kafka-pubsub.googleapis.com:443";
+  }
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply pubsublite-kafka-auth/src/main/java/com/google/cloud/pubsublite/kafka/ClientParameters.java:54]: Closed per offline discussion.

File: kafka_gcp_credentials.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        self.send_header('Content-type', 'text/plain')
+        self.end_headers()
+        self.wfile.write(build_message().encode('utf-8'))
+
+
+def run_server():
+    server_address = ('localhost', 14293)
+    server = http.server.ThreadingHTTPServer(server_address, AuthHandler)
+    print("Serving on localhost:14293. This is not accessible outside of the "
+          "current machine.")
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply kafka_gcp_credentials.py:65]: Yes, but an arbitrary fixed number that seems to not be used by anything else.

**Review by hannahrogers-google — APPROVED** [review]

**Pull request merged**